shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumTransferLeaderCommand.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.fsadmin.journal;
+
+import alluxio.cli.fsadmin.command.AbstractFsAdminCommand;
+import alluxio.cli.fsadmin.command.Context;
+import alluxio.client.journal.JournalMasterClient;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.exception.ExceptionMessage;
+import alluxio.exception.status.InvalidArgumentException;
+import alluxio.grpc.NetAddress;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Options;
+
+import java.io.IOException;
+
+/**
+ * Command for transferring the leadership to another master within a quorum.
+ */
+public class QuorumTransferLeaderCommand extends AbstractFsAdminCommand {
+
+  public static final String ADDRESS_OPTION_NAME = "address";
+
+  public static final String OUTPUT_RESULT = "Transferred leadership to server: %s";
+
+  /**
+   * @param context fsadmin command context
+   * @param alluxioConf Alluxio configuration
+   */
+  public QuorumTransferLeaderCommand(Context context, AlluxioConfiguration alluxioConf) {
+    super(context);
+  }
+
+  /**
+   * @return command's description
+   */
+  @VisibleForTesting
+  public static String description() {
+    return "Transfers leadership of the quorum to the master located at <hostname>:<port>";
+  }
+
+  @Override
+  public int run(CommandLine cl) throws IOException {
+    JournalMasterClient jmClient = mMasterJournalMasterClient;
+    String serverAddress = cl.getOptionValue(ADDRESS_OPTION_NAME);
+    NetAddress address = QuorumCommand.stingToAddress(serverAddress);
+
+    jmClient.transferLeadership(address);
+    mPrintStream.println(String.format(OUTPUT_RESULT, serverAddress));
```

Comment:
We could, this is the same logic as the remove command. The remove command also does not externally validate whether the removal was successful.

Reply:
If remove is asynchronous as well, then we should update that too.
Since you mentioned, the leader is not elected upon returning transferLeaderShip() we should verify that here, and not return until election is resulted. We should also make sure we successfully rigged the election for our candidate.